src/client/pythonEnvironments/common/externalDependencies.ts

```diff
@@ -19,18 +21,23 @@ export function initializeExternalDependencies(serviceContainer: IServiceContain
 
 // processes
 
-function getProcessFactory(): IProcessServiceFactory {
-    return internalServiceContainer.get<IProcessServiceFactory>(IProcessServiceFactory);
-}
-
-export async function shellExecute(command: string, timeout: number): Promise<ExecutionResult<string>> {
-    const proc = await getProcessFactory().create();
-    return proc.shellExec(command, { timeout });
-}
-
-export async function exec(file: string, args: string[], options: SpawnOptions = {}): Promise<ExecutionResult<string>> {
-    const proc = await getProcessFactory().create();
-    return proc.exec(file, args, options);
+export async function shellExecute(
+    command: string,
+    timeout: number,
+    defaultEnv?: EnvironmentVariables,
+    disposables?: Set<IDisposable>,
+): Promise<ExecutionResult<string>> {
+    return shellExec(command, { timeout }, defaultEnv, disposables);
+}
```

Comment:
Same here

Also please add a comment here stating that we do not need custom env variables for the discovery component, which is why we're doing this. So that we don't accidently use this code for other components.

```suggestion
export async function shellExecute(
    command: string,
    timeout: number,
    disposables?: Set<IDisposable>,
): Promise<ExecutionResult<string>> {
    // The comment goes here
    return shellExec(command, { timeout }, undefined, disposables);
}
```